### PR TITLE
#3 - User Authentication

### LINKED ISSUE #3: User registration
Registration form for new users (fields: name, email, password, user type – student, teacher, accept terms of use).

### LINKED ISSUE #4: User login
Login form with data validation.

### LINKED ISSUE #5: Password reset feature
User should be able to reset his password

### PR DESCRIPTION
**Pull request by Silabalkan**
Implemented user authentication using Laravel Breeze.
Integrated login and registration functionality.
Resolved conflicts with the authentication code.
Connected the authentication features with the Vue.js and Inertia.js front end.
This should close #3
This should close #4
This should close #5

### CODE REVIEW

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Please resolve conflicts
2. Please change pr title - we have a rule that pr title should look like this:

#issue-number - title

So in this case, we should give title like 
#3 - user auth

And also please remember that in pr description we should write down every issue that is connected with pr or any issues that should bo closed after merging this pr. Something like: This should close #issue-number

3. Please don't push `.env` file

**Review by senagokhan — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,5 @@ yarn-error.log
 /.vscode
 /.zed
 .php-cs-fixer.cache
+
+
```
Comment (EwelinaSkrzypacz) [inline .gitignore]:
```suggestion
```

**Review by EwelinaSkrzypacz — CHANGES_REQUESTED** [review]
Checks are failed - please fix this

File: .gitignore (hunk 1 above)
Comment (EwelinaSkrzypacz) [reply .gitignore]: At the end of the file should be only one empty line

File: vite.config.js [hunk 2]
```diff
@@ -35,3 +35,4 @@ export default ({ mode }) => {
         ],
     })
 }
+
```
Comment (EwelinaSkrzypacz) [inline vite.config.js]:
```suggestion
```

File: resources/js/bootstrap.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import _ from 'lodash'
+import axios from 'axios'
+
+// window için tür tanımlama
```
Comment (EwelinaSkrzypacz) [inline resources/js/bootstrap.ts]: :eyes:

If you really need this comment, please write it in English

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Please fix all warnings from GitHub - like this

image

2. At our company, we avoid leaving comments in the code unless absolutely necessary. We believe that code should be self-explanatory. That said, if you really feel a comment is needed, don’t worry—we can live with it. 😄 Let’s just keep it to a minimum.

I know that most of comments are from Laravel Breeze and fresh Laravel installation, but I think we can remove them

3. I think that tasks #4 and #5 can be closed because in Laravel Breeze out of the box we have features like login to application and reset password

4. For me reset password doesn't work, i got an error like 

`The "" scheme is not supported; supported schemes for mailer "smtp" are: "smtp", "smtps".`

I read that is problem with some composer dependency, so if you have the same problem, run `composer require "symfony/mailer:~7.1.0"` and push this - it should work

File: vite.config.js [hunk 4]
```diff
@@ -34,4 +34,4 @@ export default ({ mode }) => {
             }),
         ],
     })
-}
+}
```
Comment (EwelinaSkrzypacz) [inline vite.config.js]: Now empty line at the end of the file is missing :eyes:

File: package.json [hunk 5]
```diff
@@ -32,7 +31,7 @@
         "eslint-plugin-tailwindcss": "^3.17.4",
         "eslint-plugin-vue": "^9.28.0",
         "postcss": "^8.4.31",
-        "tailwindcss": "^3.2.1",
+        "tailwindcss": "^3.4.16",
```
Comment (EwelinaSkrzypacz) [inline package.json:33]: It should be in `dependencies` not in `devDependencies`

File: resources/js/Pages/Auth/ForgotPassword.vue [hunk 6]
```diff
@@ -21,6 +22,8 @@ const submit = () => {
 }
 </script>
 
+
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Auth/ForgotPassword.vue]: These two empty lines can be removed

File: resources/js/Pages/Auth/Login.vue [hunk 7]
```diff
@@ -29,6 +30,7 @@ const submit = () => {
 }
 </script>
 
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Auth/Login.vue]: This extra empty line can be removed

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Check is failed - please run `make fix` - it should fix all errors
2. Extra blank lines have been unnecessarily added at the end of the file, resulting in two blank lines. This clutters the code and doesn't follow standard formatting practices. Please keep only one blank line at the end.

**Comment by EwelinaSkrzypacz** [comment]
I think that tasks #4 and #5 can be closed because in Laravel Breeze out of the box we have features like login to application and reset password - so please write in pr description "This should close " and task number.

File: resources/js/Pages/Profile/Partials/UpdateProfileInformationForm.vue [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            class="text-sm text-gray-600"
+          >
+            Saved.
+          </p>
+        </Transition>
+      </div>
+    </form>
+  </section>
+</template>
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Profile/Partials/UpdateProfileInformationForm.vue]: Remove this extra empty line; please search other places where is extra empty line/lines and remove them.

File: resources/js/app.ts [hunk 9]
```diff
@@ -1,16 +1,24 @@
 import '../css/app.css'
-import { createApp, h, type DefineComponent } from 'vue'
+import './bootstrap'
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/app.ts]:
```suggestion
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
Please fix GitHub errors and warnings in `resources/js/Pages/Auth/VerifyEmail.vue`. It's just some on them :point_down: 

image

You can use `make fix` to fix errors, but this command sometimes doesn't fix all and errors must be fixed manually.

File: resources/js/Pages/Profile/Edit.vue [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        <div
+          class="bg-white p-4 shadow sm:rounded-lg sm:p-8"
+        >
+          <DeleteUserForm class="max-w-xl" />
+        </div>
+      </div>
+    </div>
+  </AuthenticatedLayout>
+</template>
+
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Profile/Edit.vue]: unnecessary empty line

**Review by EwelinaSkrzypacz — DISMISSED** [review]

**Review by selinuluca — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**